rispy/writer.py

```diff
@@ -0,0 +1,122 @@
+"""RIS Writer"""
+
+import logging
+
+from .config import LIST_TYPE_TAGS
+from .config import TAG_KEY_MAPPING
+
+__all__ = ['dump', 'dumps']
+
+
+def _inverse_mapping(mapping):
+
+    # See PR #16 (if k is not 'N2')
+    return {v: k for k, v in mapping.items() if k is not 'N2'}
+
+
+class BaseWriter(object):
+    START_TAG = None
+    END_TAG = 'ER'
+    IGNORE = []
+    PATTERN = None
+
+    def __init__(self, references, mapping, type_of_reference):
+        self.references = references
+        self.mapping = mapping
+        self._rev_mapping = _inverse_mapping(mapping)
+        self.type_of_reference = type_of_reference
+
+    def _get_reference_type(self, ref):
+
+        if "type_of_reference" in ref.keys():
+            # TODO add check
+            return ref["type_of_reference"]
+
+        if self.type_of_reference is not None:
+            return self.type_of_reference
+        else:
+            raise ValueError("Unknown type of reference")
+
+    def _format_line(self, tag, value=""):
+        """Format a RIS line."""
+        return self.PATTERN.format(tag=tag, value=value)
+
+    def _format_reference(self, ref, count):
+
+        lines = []
+
+        lines.append("{i}.".format(i=count))
+        lines.append(self._format_line(
+            self.START_TAG,
+            self._get_reference_type(ref)
+        ))
+
+        for label, value in ref.items():
+
+            # not available
+            try:
+                tag = self._rev_mapping[label.lower()]
+            except KeyError:
+                logging.info("label {} not exported".format(label))
+                continue
+
+            # ignore
+            if tag in [self.START_TAG] + self.IGNORE:
+                continue
+
+            # list tag
+            if tag in LIST_TYPE_TAGS:
+                for val_i in value:
+                    lines.append(self._format_line(tag, val_i))
+            else:
+                lines.append(self._format_line(tag, value))
+
+        lines.append(self._format_line("ER"))
+        lines.append("")
+        lines.append("")
+
+        return lines
+
+    def format(self):
+
+        for i, ref in enumerate(self.references):
+            lines_ref = self._format_reference(ref, count=i + 1)
+            for line in lines_ref:
+                yield line
+
+
+class RISWriter(BaseWriter):
+
+    START_TAG = 'TY'
+    PATTERN = '{tag}  - {value}'
+
+
+def dump(references, file, mapping=None):
+    """Dump a ris lines to file.
+
+    Entries are codified as dictionaries whose keys are the
+    different tags. For single line and singly occurring tags,
+    the content is codified as a string. In the case of multiline
+    or multiple key occurrences, the content is returned as a list
+    of strings.
+
+    Keyword arguments:
```

Comment:
revise format slightly here to more closely follow google style comments: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

It looks like you followed the prior convention, @MrTango is it ok if we progressively switch to a standard comment style?